### ISSUE TITLE
Suggested revision of <sourceDesc>

### ISSUE THREAD

**Issue by BruxDDay** [music-encoding/music-encoding#478]
Preface:

As the MEI header has become more commonly adapted to the creation of catalogs or metadata focused applications separate from files of MEI music notation encoding we see a problem with a dual interpretation of the element `sourceDesc`

The two roles can be summarized as follows:

1. The original intent of `sourceDesc` was to identify materials used to created digital music notation. This usage does not require a physical description of a specific document (FRBR Item), rather only a bibliographic identification. With this intent in mind, it makes sense to locate the element as part of the `fileDesc` element

2. The evolving use of `sourceDesc` is not linked to music notation necessarily, but mostly a physical description of a specific source/document. It may involve a very detailed physical and (and possibly other aspects) description. This use of `sourceDesc` would apply to a collection catalog. Maybe it would also apply to thematic catalog but the descriptive elements would be more selectively applied.

Suggested solutions:

1. Change the name of `sourceDesc` in the `fileDesc` to `sourceIdentifer`

2. Create a new level in the MEI Header titled `sourceDesc` independent of both `fileDesc` and `workDesc.` In this new level develop and refine elements that could be used to describe a physical document (independent of the abstract work described in `workDesc`.)

3. Place the new `sourceDesc` category after `workDesc` in the MEI hierarchy. In this sense the new`sourceDesc` will correspond with the "Item" level of a FRBR description. Manuscript sources will be unique by definition, and detailed descriptions of specific print sources would also be unique and correspond with "Item" in the FRBR hierarchy. This new use of `sourceDesc` might require some rethinking of `@analog` to describe FRBR relationships.

4. We further suggest that `revisionDesc` should be relocated as a sub-element of `fileDesc` because it is intended to describe revisions to MEI code, not musical revisions in a specific document. It would also be good to rename the current `revisionDesc` to `codeRevisionDesc` in order to avoid any confusion with a future use of `revisionDesc` in the new category of `sourceDesc`

**Comment by bwbohl** [comment]
I've tried to tackle this huge issue a couple of times now… but still don't manage to address everything. Something that might be connected to this is #61 (allow `<bibl>` in `<source>` as an alternative to the structured elements)

**Comment by BruxDDay** [comment]
I wonder if this issues might also be related indirectly to Dennis Reid's post to the MEI list on December 1st. At least in principle it seems wrong to use the current `sourceDesc` for the creation of catalogs because the creation of a catalog requires much more flexibility in describing sources than was ever intended for the simple identification of the source used to create an encoded notation file. The creation of a separate `sourceDesc` intended specifically for describing sources in different kinds of catalogs seems to be a necessity.

**Comment by pe-ro** [comment]
`<sourceDesc>` need not be limited to the sources _**actually used**_ in the creation of the file -- one could think of it as a container for description of _**source material generally**_.  The actual sources employed as witnesses are/should be identified using `@source` on `<lem>` or `<rdg>`.

**Comment by ndubo** [comment]
I agree that `<sourceDesc>` (as child of `<fileDesc>`) should be limited to the "sources actually used in the creation of the file". But we have to acknowledge the fact, that there are many use cases where no 'file' (in the sense of coded music) is created, mainly work catalog entries and stand-alone source descriptions that rely on the MEI header by itself. 
I do not think that a seperate `<sourceDesc>` "intended specifically for describing sources" should be created; it's up to the user how simple or how deep he/she wants to describe the sources. But in order to reconcile the various use cases the current `<sourceDesc>` could be moved out of the `<fileDesc>` and hierarchically coordinated with `<fileDesc>` and `<workDesc>` where it may receive additional features in order to make it more flexible.   
Within the `<filesDesc>` a `<sourceIdentifier>` as suggested in @BruxDDay's original proposal could be used as pointer to the corresponding item in the `<sourceDesc>.`

**Comment by pe-ro** [comment]
@ndubo, I see the logic in the approach, I'm just trying to find the best path.

We need to maintain backward compatibility as much as possible, which would suggest leaving `<sourceDesc>` in place to hold descriptions of the "sources used in the creation of the file" while also allowing it to use its `@target` attribute to point into a catalog of a more comprehensive nature.  This would not only maintain compatibility, but let those who want to describe the source material "in place" to continue to do so while also accommodating those want to use a pointer instead.

The target of the pointer could reside within a new `<workList>` or `<workCatalog>` element.  This element can take over the descriptive functions now covered by the combination of `<work>`, `<expression>`, `<source>`, and `<item>`.   `<workList>` can be allowed to be a document-level element so that catalogs may be placed in external files.   In fact, it could be a new, ancillary MEI document type; i.e., it wouldn't occur in the header at all, making for a cleaner separation between metadata for the current encoding (captured within `<meiHead>`) and an independent work list/catalog.

**Comment by BruxDDay** [comment]
I like this approach. The ability to create a catalog as a separate MEI document type suggests to me greater freedom in creating and managing different elements and attributes for advanced description of rare and manuscript sources (if needed). This approach should also maintain backward compatibility, which I recognize is very important. I am assuming that within a new `<workList>` document it would still be possible to use all the MEI notation features for creation of incipits. I am also thinking that the FRBR elements may still play a role in a separate `<workList>` document, depending on the nature of the list or catalog. Are these two assumptions correct?

**Comment by bwbohl** [comment]
I understand the logic and appeal of sth. like `<workList>` or `<workCatalog>` but from my perspective, proposing a model for cataloging is not what MEI is really about. Our focus should rather be how to use MEI encoded information in accordance with catalogs. I bet there are many models for catalogs out there somewhere, and of course a format for cataloging should not "be limited" to music.

The approaches I encountered in various music edition projects (in which a catalog of works, sources, or other material is of course an intended output) was rather on a file-per-object basis (with an object being something that would be defined by the project).

Having consumed FRBR to some degree (and of course in the general light of semantic web) and of course taking in @pe-ro concern of backwards compatibility, a good approach should allow both:

1. handling all there is to say about the work, source, musical text in one comprehensive file (see above comments)
2. having separate files for each defined entity (see below)

### entity based MEI

The Edirom Online software for example has a couple of files when encoding a music edition:
a) an edition.xml (custom schema non MEI)
b) a work.xml (MEI, describing frbr:work- and grbr:expression-level metadata and including the edition's critical notes)
c) any number of source.xml files describing fror:manifestation- and fror:item-level metadata

Nevertheless this approach is not fully modelled to all extents as the development approach of Edirom generally is to allow any kind of MEI and not to specify how and where to encode what information.

This approach could be extended to some merit by having separate files for the primary for-entities:

* frbr:work.xml
* frbr:expression.xml
* frbr:manifestation.xml
* frbr:item.xml

Doing so would also allow a better granularity in data exchange e.g. with memory institutions like libraries or authority file providers. While four my perspective a all-in-one file approach would not!

### conclusion

Generating a work catalog on the basis of such a data-set would be a higher-level operation that, yes could be done in MEI e.g. using `<meiCorpus>` or rather any combination of presentation-layer query language an markup language to render some nice web or print output.

So it's -1 for `<workCatalog>` from me.
And -1 for `<workList>`. Though it could be interesting in describing a source's contents this can be handled be frbr:relations, already.

**Comment by kepper** [comment]
Hi all,

Like Benni, I'm reluctant to go there with MEI itself. For me, this sounds more like another MEI profile for a specific use case, but not like a new model within MEI. FRBR is extremely helpful in an editorial context, and we shouldn't give that one up in order to simplify handling of catalogues. Let's revise the problems that have been raised: 

1) //meiHead/workDesc vs. //meiHead/fileDesc/sourceDesc seems imbalanced from a FRBR perspective
2) //sourceDesc serves a double purpose (specify the materials used in preparation of the file vs. frbr:manifestion) 

To be honest, both aspects are no showstoppers for MEI, and better documentation seems to be a potential solution – not necessarily the best, but also not the worst. If we take backwards-compatibility as an argument, it's hard to get any better, though ;-)

But let's have a closer look. The main problem of 2) is that we're mixing two different meaning of "source". In the original meaning, this was intended to describe both editorially relevant prints or manuscripts, but also the arbitrary Finale or MusicXML file that was used on the way towards MEI. This particular meaning isn't covered by a common understanding of FRBR, and indeed it seems to be sensible to not conflate that in an MEI file with contents we put under the frbr:manifestation hat. From within an //app/rdg/@source, we surely wouldn't point to the description of a Finale file, but only to manuscripts etc. I suspect that the latter is the most common use of mei:source anyway, with only very few exceptions. 

What I could imagine is to promote the current /sourceDesc up to be a direct child of /meiHead. In it's current place, we could introduce a new element `<fileSource>`, which would get a stripped down version of the existing `<source>`. (Of course the name is just a straw man, and we need a container as well…). For manuscript sources, we could introduce a standardised pointer to a separate `<source>` element in the FRBR sense, where all the bibliographic details and the description of this print (etc.) could live (maybe using a `<sourceIdentifier>` as proposed by @BruxDDay, maybe using an attribute). That way, the fileSource element would just say that this source was used to prepare the file, but other sources might be described as well. 

I am fully aware that this
* causes a certain markup overhead
* isn't fully backwards compatible (though it should be fine to convert *most* existing files into this new model in a straight-forward way)
* seems to focus on music content (if encoding a catalogue only, one could assume that there has to be a fileSource for every source anyway)

However, I see the following benefits:
* It becomes possible to use the existing meiHead start element for catalogue files without introducing a new, separate model
* The new model makes no difference between catalogues, editions, or other use cases of MEI, providing the same possibilities in all situations
* This should simplify interchange between different use cases for or profiles of MEI
* It focusses on a separation of the current double meaning of source (number 2) above), and addresses 1) at the same time


I think we should seek a solution that introduces only changes that are really necessary, and are justifiable for all different use cases of MEI.

**Comment by BruxDDay** [comment]
@bwbohl 's suggestion that MEI is not primarily about catalogs, but rather music notation is well taken. However, MEI _is_ being used quite extensively for catalogs, especially thematic catalogs and catalogs of closed discrete collections that require or merit advanced descriptions. From that perspective it is important to ensure that when MEI is used for cataloging it is accommodating to the needs of specialized catalogs. The MEI use for catalogs should not be a secondary consideration. The ability to describe a rare edition or manuscript in great detail is one of the persuasive arguments for using MEI rather than MARC. Admittedly TEI also has the ability to handle advanced descriptions, but because incipits are often very important components of such catalogs, why not stay within MEI?

Could I ask for some friendly clarification on what is meant by the creation of a catalog on a file-per-item basis? Does this mean as outlined in the Edirom Edition example that separate files be created for each FRBR level of a catalog? I am not sure I follow the logic of creating separate XML files for one catalog. Please help me better understand your point.

**Comment by bwbohl** [comment]
@BruxDDay I’m absolutely with you that

> The MEI use for catalogs should not be a secondary consideration

And I never intended to be understood else!

> some friendly clarification on what is meant by the creation of a catalog on a file-per-~~item~~**object** basis

I just called it this way, as I had no better term for it ;-) And yes, you're absolutely right; the idea behind it is creating separate files for the objects (items, entries, name it…) of one catalog. In fact, when looking at a catalog in a more abstract way, it could be seen as a (curated) data collection! This makes it comparable to a database (having defined fields for each object/item). The fact that it might be printed or available in digital form is nothing more than a form of manifestation.

Let's imagine you have a printed catalog and want to create a web-version of it. You could follow two strategies:
 
-  a) sit down, create a (single) file in your computer program, open the book, turn to page one, and transcribe, and transcribe, and transcribe all the details the catalog provides on each item into one file. The growth of your data would be slow in terms of width (items covered) but constantly high in terms of depth (detail of the items' description). Your data will be usable when you've finished transcribing.

-  b) sit down, cut the book into pieces corresponding to the single items, cut the items into more pieces and put those on piles ordered by level of detail – ranging from general info e.g. work title, composer, etc. to very special information e.g. critical notes from an edition. Now you pick the "general information pile" containing all the general information of all items covered, create a file and transcribe the general information of item one, then create a file and transcribe the general information of item two, … The growth of your data would now be fast in terms of width (items covered) and constantly slow in terms of depth (detail). But your data will achieve an early stage of low detail completeness, allowing for first usage of the data.

Why separate files? Their changes can be controlled more easily and a change on one file would reflect a change on one item. While a change on your huge file would reflect a change on any one item in the catalog or on several.

**Comment by kepper** [comment]
From my perspective, this aspect is surely interesting, but a separate issue. MEI utilises the anyURI datatype, so it is equally possible to put everything into one file or split it up into separate files for each FRBR entity. The only missing thing here would be to allow <work>, <expression>, <source> and <item> as start elements, but again, that seems to be a separate discussion. I think we should leave it to individual projects to decide what approach is more practical to them. MEI should have a way to keep everything in one file, and be it only for better interchange or archiving purposes. As I mentioned in Mainz, for me the boundaries between data and metadata are increasingly blurred, and I believe that this is a good thing, as it (potentially) combines the strengths of both worlds. Admittedly, this requires good guidance, but this is what this discussion is about, right?

**Comment by pe-ro** [comment]
@BruxDDay, both your assumptions are correct -- Existing features of MEI, such as `<incip>`, would be available within `<workList>`.  Also, `<workList>` would make use of FRBR, though its use wouldn't be required.

**Comment by axgeertinger** [comment]
In my opinion, separating catalog (meta-)data from the notation encodings would be a great step backwards. Apart from the fact that the MEI header is capable of handling far more detailed information than any cataloging system I know, my motivation for using MEI for cataloging projects has always been a vision of breaking down the boundaries between editions and catalogs. I acknowledge the different and to some degree conflicting intentions of the two current uses of `<sourceDesc>`, but I think @kepper's suggestion is much better than introducing a `<workList>` or `<workCatalog>`. 
I see it this way: A music encoding should always include some amount of metadata, but the scope of these metadata may vary widely. Extending this information to the information given in a (thematic) catalog is to me basically a question of degree, and ultimately I would like to see an edition and a catalog to be only two different perspectives on the same pile of data. In  our Nielsen catalog, for instance, we may want to include the entire Nielsen edition encoded in MEI at some point. We wouldn't have to generate separate incipits but could then simply show the first page of the scores - but as the entire music would also be encoded, the user could read through the rest of the score as well. Would that be a catalog or an edition? Perhaps both. So, I'd definitely prefer keeping things together.

**Comment by pe-ro** [comment]
@axgeertinger expertly articulates our common goals.  I'm simply looking for a more efficient, better all-around methodology for achieving them.

The amount of change brought about by this proposal is minimal.  The number of changes to the schema is comparable to those suggested by @kepper.  I believe, however, that the `<workList>` approach does a better job of cleaning up the current confusion over `<source>` while also providing a convenient way to separate in-depth from light-weight metadata when that's desirable.

`<workList>` is intended to be a refinement of `<workDesc>`, which already allows for the description of multiple works.  No encoding options would be lost, only moved.  That is, the data found in `<source>` would also be allowed within `<work>`, although likely renamed to `<manifestation>`.  This will make it possible to encapsulate the WEMI (work, expression, manifestation, item) description within `<work>` for those who desire to follow this approach.  `<source>` will retain its current content model to satisfy backward compatibility and simple use cases, but can also be used to reference data captured in `<manifestation>` to avoid duplication.

`<workList>` will be allowed to be a document element for those cases where it makes sense not to duplicate the WEMI information across multiple encodings.  Something like this capability has always been possible using XInclude, but under this plan a separate workList file could be validated independently of the file into which it's included.

Since `<work>`, `<expression>`, `<source>`, and `<item>` will retain their current child elements, the level of cataloging detail will remain the same and still be variable according to local needs.

`<work>` already allows `<incip>`, which may in turn contain an MEI-encoded incipit, one in another encoding scheme, or a pointer to a graphic, so no stumbling blocks are introduced by this proposal with regard to plans for the Nielsen catalog or any other.  I've long held that it would be beneficial to provide an XPath-based pointing mechanism on/within `<incip>` to allow it to identify an excerpt within the transcription.  Even so, the presence or function of `<incip>` has very little to do with the current proposal.

Early next week, I plan to provide an ODD customization implementing the proposed changes so that the discussion can become more concrete than theoretical. 😀

**Comment by pe-ro** [comment]
> as the entire music would also be encoded, the user could read through the rest of the score as well. Would that be a catalog or an edition?

In my opinion, it's both.  But in reaching for that outcome, we shouldn't limit the possibility of separation of these two options.  One should be able to use MEI to encode an edition **_or_** a catalog as necessary without combining them.

**Comment by pe-ro** [comment]
Most of this issue is addressed by cd39884.  Only the question of whether to provide a `<workList>` element remains.  I'd prefer to replace `<workDesc>` with `<workList>` --
```
<elementSpec ident="workList" module="MEI.shared" mode="add">
  <desc>A list of musical works.</desc>
  <classes>
    <memberOf key="att.bibl"/>
    <memberOf key="att.basic"/>
    <memberOf key="att.labelled"/>
    <memberOf key="att.lang"/>
    <memberOf key="att.meiVersion"/>
  </classes>
  <content>
    <rng:zeroOrMore>
      <rng:ref name="model.headLike"/>
    </rng:zeroOrMore>
    <rng:zeroOrMore>
      <rng:choice>
        <rng:ref name="model.textComponentLike"/>
      </rng:choice>
    </rng:zeroOrMore>
    <rng:oneOrMore>
      <rng:choice>
        <rng:ref name="model.workLike"/>
        <rng:ref name="workList"/>
      </rng:choice>
    </rng:oneOrMore>
    <rng:zeroOrMore>
      <rng:choice>
        <rng:ref name="model.textComponentLike"/>
      </rng:choice>
    </rng:zeroOrMore>
  </content>
</elementSpec>
```
which has the same basic model, but allows leading and trailing text components and nests recursively.

When used inside `<meiHead>`, `<workList>` should function in the same way as `<workDesc>`; i.e., describe only those works that are/will be encoded in the containing MEI file.  When used as a separate document, however, `<workList>` could function as a work catalog.

**Comment by pe-ro** [comment]
Closing this issue for now.
